core/src/view/view.cpp

```diff
@@ -105,7 +105,13 @@ void View::setPitch(float _pitch) {
 
 void View::translate(double _dx, double _dy) {
 
-    setPosition(m_pos.x + _dx, m_pos.y + _dy);
+    auto mapBounds = m_projection->MapBounds();
+    glm::dvec2 newMapPos(
+                        glm::clamp(m_pos.x + _dx, mapBounds.min.x, mapBounds.max.x),
+                        glm::clamp(m_pos.y + _dy, mapBounds.min.y, mapBounds.max.y)
+                        );
+
+    setPosition(newMapPos.x, newMapPos.y);
```

Comment:
Any logic applied in `translate()` should also be applied in `setPosition()`, so the logic should reside in `setPosition()` and then `translate()` can just blindly pass the incremented coordinates (like `zoom()` does).